### DISCUSSION TITLE
Current release status?

### DISCUSSION THREAD

**Discussion by SilentSin** [statiqdev/Discussions#3] [General]
I'll be updating the documentation for a major update of my project soon, which would be the perfect time to migrate from Wyam to Statiq as well. So I have two questions:

1. Is the current release solid enough to generate a documentation site roughly as good as Wyam or does it still have some significant missing features?
2. Can you give an estimated release date for the end of the Beta? Next week? Next month? Longer?

**Comment by daveaglick** [comment]
Great question, I know lots of folks have been wondering the same. While I don't have any specific timelines, I can give a general update on where things stand.

**Statiq Framework** is very close to a 1.0 release. At this point I'm mainly keeping it in beta just in case something big comes up while working on porting over documentation generation from Wyam. I don't anticipate any big changes as a result of that, but you never know.

**Statiq Web** is maturing rapidly (though now with summer upon us in the US my time has been cut a little and the cadence has slowed a bit). There's still a few things I'd like to get in feature-wise to reach parity with the old Wyam blog recipe, but we're getting close. Themes is the big missing link and I'm planning on starting on those any day now. After the first couple are done I think they'll flow pretty quickly, especially with the way Statiq pushes _all_ functionality into the theme (unlike Wyam which required certain conventions to be honored).

**Statiq Docs** hasn't been started yet, though an enterprising @mholo65 has been using the relevant bits from Statiq Framework to generate API docs so I've got some confidence there aren't any big unknown blockers lurking in the dark. My best guess at a timeline for starting Statiq Docs and porting the docs theme is late summer.

Hopefully that gives a general overview. For your specific questions:

> Is the current release solid enough to generate a documentation site roughly as good as Wyam

In many, many ways Statiq is _way_ better and more solid. I certainly think it's better architected and likely has less lingering bugs. That said, until work begins on Statiq Docs you'd need to add your own pipeline that uses the `AnalyzeCSharp` module and write layouts and pages that make use of the output. So it's definitely not fire-and-forget yet for API docs.

> Can you give an estimated release date for the end of the Beta? Next week? Next month? Longer?

It's a fair question, especially in the context of planning work - and I know you've been a heavy user and contributor to Wyam so I'm invested in doing what I can to help. My best guess at a Wyam-like experience to ship is very late summer/early fall. A lot of the pieces are in place so I'm hopeful it'll just start clicking once work begins. That said my estimates tend towards the optimistic, so perhaps September/October is a better estimate for docs betas.

**Comment by SilentSin** [comment]
Thanks, that answers my questions nicely.

One thing that might be worth adding to your FAQ is a section about why we should upgrade from Wyam, because it sounds like there are going to be some good reasons.

**Reply by daveaglick** [reply]
Great idea - added https://github.com/statiqdev/statiqdev.github.io/issues/11 to get that into the docs

**Comment by TechWatching** [comment]
Hi I am looking for upgrading from Wyam to Statiq Web. I have seen that Statiq.Web is still not in GA. So I just wanted to know :
- when is it going to be GA
- what are the things missing or preventing Statiq Web to be GA ?
- are there any reason I should know not to migrate yet and wait until it is GA

Thanks :)

**Reply by devlead** [reply]
So I'll let @daveaglick answer any GA questions but I suspect it's when it's done.

But I can give my input on the two other points as someone who's migrated a few sites to Statiq.

> - what are the things missing or preventing Statiq Web to be GA ?

In my opinion, the only major piece missing from WYAM is the Docs parts, work has started on it and .NET assembly docs still are totally doable but require you to do some plumbing yourself. 


> - are there any reason I should know not to migrate yet and wait until it is GA

For me, the benefits by far outweigh the drawbacks, the potential drawbacks as I see it are:

- Breaking API changes
- Breaking behaviors

In my experience these have been very few apart, the latest betas have been really solid, and any breaking changes have been mentioned in the release.

An advantage of a static site generator is that as long as you're happy with the output you're fine.

Once migrated it's been for me a joyful experience, better perf, easier to extend and understand, as it's now an "regular" .NET Console application, with migrating mileage might vary for how hard it is to migrate depending on how much you've customized your WYAM site, but i.e. migrating my blog was an evening's effort, it took a little more time as I at the same time customized the site and added a few features I didn't have before, but that was not Statiqs "fault, but me getting carried away and inspired to do so.

**Reply by daveaglick** [reply]
> when is it going to be GA

I wish I could give an answer to this, but I've learned over the years not to :). I try hard and have learned through experience to strike a good live/work/oss balance to avoid burnout. That often means stretches (sometimes long ones) with minimal open source work. I'm actually coming out of one right now over the summer. Because of that it's very difficult to predict with any kind of certainty when certain milestones will be met - I know other open source projects manage it, but this unfortunately isn't one that can. What I will say is that project continues to move in the right direction and _will_ make GA eventually - hopefully sooner rather than later.

> what are the things missing or preventing Statiq Web to be GA

> In my opinion, the only major piece missing from WYAM is the Docs parts, work has started on it and .NET assembly docs still are totally doable but require you to do some plumbing yourself

Yep, this is exactly the hold-up. In my mind, Statiq Web is production ready today after having been in use for almost 2 years. There are still some features I'd love to add, though I'll likely add those post version 1.0 since they shouldn't be breaking. The thing holding back a 1.0 release though is that I'm still working on Statiq Docs (which is built on Statiq Web). I don't anticipate any big breaking changes due to that, but want to reserve the ability to make them for Statiq Docs if I really need to. Once Statiq Docs is ready, I'll release all three projects as GA at the same time.

> are there any reason I should know not to migrate yet and wait until it is GA

I can't think of any. There are some substantial breaking changes (though [they're mostly documented](https://www.statiq.dev/web/porting-from-wyam)) and I don't anticipate any more - so work you do now to migrate will stay relevant when a 1.0 is launched. For that matter, I don't expect much fanfare from a 1.0 release anyway - it'll likely just be me ticking over the version from whatever beta was before it and will be a more marketing change than technical one.

**Comment by SilentSin** [comment]
What's the current state of Statiq Docs relative to Wyam? Is now a good time to make the switch? If not, do you have any sort of roadmap or checklist of things you plan to work on leading up to the official release of 1.0?

I don't have any custom pipelines or anything, I'm just wondering if the default API pages are currently as good in Statiq as Wyam (class diagrams, page formatting, member lists, etc.).

Also, this GitHub feature is nifty:
![image](https://user-images.githubusercontent.com/9582550/178091707-b818d16b-9268-470d-b482-8f07baaae1cf.png)
I used it and found ... my old thread.

**Reply by daveaglick** [reply]
> What's the current state of Statiq Docs relative to Wyam?

So close! The "engine" bits are basically done. That's not to say I won't keep tinkering, but the `AnalyzeCSharp` module does everything the Wyam one did and a little more.

There's still a little work to be done on the theme. It looks pretty much like the Wyam one does, at least once you get inside the layout portions (the layout itself is a little newer and nicer). The Statiq site is now generating it's own API docs as an example: https://www.statiq.dev/api/. The gaps are in some features that I have yet to port over like API searching - those are cataloged in the theme issues at https://github.com/statiqdev/Docable/issues.

> I used it and found ... my old thread

Lol. Searching for something I have no idea how to do, and then finding a 5 year old blog post of mine explaining exactly how to do the thing I have no idea how to do happens to me not infrequently.

**Reply by daveaglick** [reply]
And then of course there's a backlog full of tweaks, edge cases, and bugs that were likely inherited from Wyam. I need to do through all those older Wyam docs issues and triage them into either the Statiq Docs or the Docable (theme) repo so I can get back to addressing them.
